systemd/sdnotify_test.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package systemd_test
+
+import (
+	"net"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/systemd"
+)
+
+type sdNotifyTestSuite struct{}
+
+var _ = Suite(&sdNotifyTestSuite{})
+
+func (sd *sdNotifyTestSuite) TestSdNotifyMissingNotifyState(c *C) {
+	c.Check(systemd.SdNotify(""), ErrorMatches, "cannot use empty notify state")
+}
+
+func (sd *sdNotifyTestSuite) TestSdNotifyWrongNotifySocket(c *C) {
+	for _, t := range []struct {
+		env    string
+		errStr string
+	}{
+		{"", "cannot find NOTIFY_SOCKET environment"},
+		{"xxx", `cannot use NOTIFY_SOCKET "xxx"`},
+	} {
+		os.Setenv("NOTIFY_SOCKET", t.env)
+		defer os.Unsetenv("NOTIFY_SOCKET")
+
+		c.Check(systemd.SdNotify("something"), ErrorMatches, t.errStr)
+	}
+}
+
+func (sd *sdNotifyTestSuite) TestSdNotifyIntegration(c *C) {
+	for _, sockPath := range []string{
+		filepath.Join(c.MkDir(), "socket"),
+		"@socket",
+	} {
+		os.Setenv("NOTIFY_SOCKET", sockPath)
```

Comment:
Because I'm allergic to setenv I'd rather have us do something like this
```
type SdNotify struct {
  SocketPath string
}

func (sn *SdNotify) Notify(msg string) error {
 ...
}
```

Then you can test this easily and you can skip calling the notify function if `Path` is empty or something similar.